crawl-ref/source/god-abil.cc

```diff
@@ -2166,88 +2167,101 @@ void cheibriados_time_step(int pow) // pow is the number of turns to skip
     mpr("You return to the normal time flow.");
 }
 
-bool ashenzari_transfer_knowledge()
-{
-    if (you.transfer_skill_points > 0 && !ashenzari_end_transfer())
-        return false;
-
-    while (true)
-    {
-        skill_menu(SKMF_RESKILL_FROM);
-        if (is_invalid_skill(you.transfer_from_skill))
+/**
+ * Choose skills to boost accompanying the current curse.
+ */
+static void _choose_curse_knowledge()
+{
+    // This loop choses two available skills without replacement,
+    // it is the two element version of "The Grunt Algorithm" used elsewhere
+    // in crawl-code to pick from a set of unknown size without building an
+    // explicit list.
+    //
+    // If Ashenzari curses need some fancier weighting this is the
+    // place to do that weighting.
```

Comment:
We should talk about the pros and cons of this in ##crawl-dev (tomorrow, when I'm awake). I can see arguments either way, and I suspect players will have strong opinions.